providers/base/tests/test_stress_ng_test.py

```diff
@@ -0,0 +1,292 @@
+    @patch("stress_ng_test.sys.argv", ["stress_ng_test.py", "cpu"])
+    def test_main_stress_ng_not_found(
+        self, shutil_which_mock, os_geteuid_mock, check_output_mock
+    ):
+        self.assertEqual(main(), 1)
+
+    @patch("stress_ng_test.check_output", side_effect=TimeoutExpired(b"", 1))
+    @patch("stress_ng_test.os.geteuid", return_value=0)
+    @patch("stress_ng_test.shutil.which", return_value="/usr/bin/stress-ng")
+    @patch("stress_ng_test.sys.argv", ["stress_ng_test.py", "cpu"])
+    def test_main_timeout(
+        self, shutil_which_mock, os_geteuid_mock, check_output_mock
+    ):
+        self.assertEqual(main(), 1)
+
+    @patch("stress_ng_test.check_output", side_effect=KeyboardInterrupt)
+    @patch("stress_ng_test.os.geteuid", return_value=0)
+    @patch("stress_ng_test.shutil.which", return_value="/usr/bin/stress-ng")
+    @patch("stress_ng_test.sys.argv", ["stress_ng_test.py", "cpu"])
+    def test_main_keyboard_interrupt(
+        self, shutil_which_mock, os_geteuid_mock, check_output_mock
+    ):
+        self.assertEqual(main(), 1)
+
+    @patch(
+        "stress_ng_test.check_output",
+        side_effect=CalledProcessError(1, b"", b""),
+    )
+    @patch("stress_ng_test.os.geteuid", return_value=0)
+    @patch("stress_ng_test.shutil.which", return_value="/usr/bin/stress-ng")
+    @patch("stress_ng_test.sys.argv", ["stress_ng_test.py", "cpu"])
+    def test_main_stress_ng_error(
+        self, shutil_which_mock, os_geteuid_mock, check_output_mock
+    ):
+        self.assertEqual(main(), 1)
+
+    @patch("stress_ng_test.check_output")
+    @patch("stress_ng_test.os.geteuid", return_value=0)
+    @patch("stress_ng_test.shutil.which", return_value="/usr/bin/stress-ng")
+    @patch("stress_ng_test.sys.argv", ["stress_ng_test.py", "cpu"])
+    def test_main_stress_cpu_success(
+        self, shutil_which_mock, os_geteuid_mock, check_output_mock
+    ):
+        self.assertEqual(main(), 0)
+
+    @patch("stress_ng_test.check_output")
+    @patch("stress_ng_test.num_numa_nodes", return_value=1)
+    @patch("stress_ng_test.swap_space_ok", return_value=True)
+    @patch("stress_ng_test.os.geteuid", return_value=0)
+    @patch("stress_ng_test.shutil.which", return_value="/usr/bin/stress-ng")
+    @patch("stress_ng_test.sys.argv", ["stress_ng_test.py", "memory"])
+    def test_main_stress_memory_success(
+        self,
+        shutil_which_mock,
+        os_geteuid_mock,
+        swap_space_ok_mock,
+        num_numa_nodes_mock,
+        check_output_mock,
+    ):
+        self.assertEqual(main(), 0)
+
+    @patch("stress_ng_test.check_output")
+    @patch("stress_ng_test.num_numa_nodes", return_value=2)
+    @patch("stress_ng_test.swap_space_ok", return_value=True)
+    @patch("stress_ng_test.os.geteuid", return_value=0)
+    @patch("stress_ng_test.shutil.which", return_value="/usr/bin/stress-ng")
+    @patch("stress_ng_test.sys.argv", ["stress_ng_test.py", "memory"])
+    def test_main_stress_memory_more_numa_nodes(
```

Comment:
Failing locally, missing `os.remove` mock?

Reply:
Resolved in b9cbbcc